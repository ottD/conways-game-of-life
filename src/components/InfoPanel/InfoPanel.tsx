import * as React from 'react';
import { Panel, Link, PrimaryButton } from 'office-ui-fabric-react';
import { useConstCallback } from '@uifabric/react-hooks';

export interface IInfoPanelProps {
  onDismiss(): void;
}

export const InfoPanel = ({ onDismiss }: IInfoPanelProps) => {

  const onRenderFooterContent = useConstCallback(() => (
    <PrimaryButton onClick={onDismiss}>Close</PrimaryButton>
  ));

  return (
    <Panel
      headerText="Conway's Game of Life"
      isOpen={true}
      onDismiss={onDismiss}
      closeButtonAriaLabel="Close"
      onRenderFooterContent={onRenderFooterContent}>
      <p>
        <Link href="https://en.wikipedia.org/wiki/Conway%27s_Game_of_Life" target="_blank">Conway`&apos;`s Game of Life</Link>
        {' '} is a cellular automaton invented devised by the British mathematician John Horton Conway in 1970.
      </p>
      <p>
        <strong>Rules:</strong>
        <ul>
          <li>Any live cell with two or three live neighbors survives.</li>
          <li>Any dead cell with three live neighbors becomes a live cell.</li>
          <li>All other live cells die in the next generation. Similarly, all other dead cells stay dead.</li>
        </ul>
      </p>
    </Panel>
  );
};