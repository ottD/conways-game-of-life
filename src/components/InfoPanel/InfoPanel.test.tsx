import * as React from 'react';
import { InfoPanel } from './InfoPanel';
import EnzymeToJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import { PrimaryButton } from 'office-ui-fabric-react';

describe('InfoPanel snapshot', () => {
    it('renders correctly', () => {
        const component = shallow(<InfoPanel onDismiss={()=>undefined}/>);
        expect(EnzymeToJson(component)).toMatchSnapshot();
    });
})

describe('InfoPanel functionality', () => {
    it('dismisses info panel', async () => {
        const onDismiss = jest.fn();
        const component = mount(<InfoPanel onDismiss={onDismiss}/>);

        expect(component.find(PrimaryButton)).toHaveLength(1);
        component.find(PrimaryButton).simulate('click');

        expect(onDismiss).toHaveBeenCalled();
        component.unmount();
    });
})