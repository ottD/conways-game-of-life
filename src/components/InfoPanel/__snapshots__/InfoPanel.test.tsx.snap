// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPanel snapshot renders correctly 1`] = `
<StyledPanelBase
  closeButtonAriaLabel="Close"
  headerText="Conway's Game of Life"
  isOpen={true}
  onDismiss={[Function]}
  onRenderFooterContent={[Function]}
>
  <p>
    <StyledLinkBase
      href="https://en.wikipedia.org/wiki/Conway%27s_Game_of_Life"
      target="_blank"
    >
      Conway's Game of Life
    </StyledLinkBase>
     
     is a cellular automaton invented devised by the British mathematician John Horton Conway in 1970.
  </p>
  <strong>
    Rules:
  </strong>
  <ul>
    <li>
      Any live cell with two or three live neighbors survives.
    </li>
    <li>
      Any dead cell with three live neighbors becomes a live cell.
    </li>
    <li>
      All other live cells die in the next generation. Similarly, all other dead cells stay dead.
    </li>
  </ul>
</StyledPanelBase>
`;
